File: Book My Show/ClientApp/src/app/Models/Movie.ts
import { Cast } from "./Cast";
import { CertificateType, Gener, Language, MovieType } from "./Enum";

export interface Movie{
        id : number
        name : string
        movieImageUrl : string,
        posterImageUrl : string,
        casts : Array<Cast>,
        languague : number,
        gener : number,
        movieType : number,
        certificateType : number
        description : string
        releaseDate : Date
        numberOfLikes : number
        movieLength : number
}  

File: Book My Show/Models/DataModels/Movie.cs
using System;
using System.Collections.Generic;
using System.Text;
using Models.Enums;
using NPoco;

namespace Models.DataModels
{
    [TableName("Movies")]
    [PrimaryKey("Id", AutoIncrement = true)]
    public class Movie
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string MovieImageUrl { get; set; }
        public string PosterImageUrl { get; set; }
        public int Language { get; set; }
        public int Gener { get; set; }
        public int MovieType { get; set; }
        public int CertificateType { get; set; }
        public string Description { get; set; }
        public double MovieLength { get; set; }
        public DateTime? ReleaseDate { get; set; }
    }
}
